panel/src/config/fiber.js

```diff
@@ -0,0 +1,373 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import Vue from "vue";
+import clone from "../helpers/clone.js";
+import debounce from "../helpers/debounce.js";
+import { merge } from "../helpers/object.js";
+import { toJson } from "../api/request.js";
+
+const Fiber = {
+  page: null,
+  swap: null,
+
+  init({ page, swap }) {
+    // callback function which handles
+    // swapping components
+    this.swap = swap;
+
+    // add the hash to the url if it exists
+    page.$url += window.location.hash;
+
+    // set initial page
+    this.setPage(page);
+
+    // back button event
+    window.addEventListener("popstate", this.onPopstateEvent.bind(this));
+
+    // remember scroll position
+    document.addEventListener("scroll", debounce(this.onScrollEvent.bind(this), 100), true);
+  },
+
+  component(name) {
+    return Vue.component(name);
+  },
+
+  async onPopstateEvent(event) {
+    // if a state is included, set the page
+    // based on this state (which will cause
+    // a swap of components)
+    if (event.state !== null) {
+      return this.setPage(event.state, { preserveState: false });
+    }
+
+    // otherwise, just make sure to update
+    // the state properly
+    const url = this.toUrl(this.page.$url);
+
+    // add the hash if it exists
+    url.hash = window.location.hash;
+
+    this.state({ ...this.page, url: url.href });
+    this.resetScroll();
+  },
+
+  onScrollEvent(event) {
+    if (
+      typeof event.target.hasAttribute === "function" &&
+      event.target.hasAttribute("scroll-region")
+    ) {
+      this.saveScroll();
+    }
+  },
+
+  data(data) {
+
+    // Add data to the Vue prototype
+    // and window.panel object if the
+    // key exists. Otherwise take from
+    // the existing window.panel object
+    [
+      "$areas",
+      "$config",
+      "$language",
+      "$languages",
+      "$license",
+      "$multilang",
+      "$permissions",
+      "$system",
+      "$translation",
+      "$urls",
+      "$user",
+      "$view"
+    ].forEach(key => {
+      if (data[key] !== undefined) {
+        Vue.prototype[key] = window.panel[key] = data[key];
+      } else {
+        Vue.prototype[key] = data[key] = window.panel[key];
+      }
+    });
+
+    // set the lang attribute according to the current translation
+    if (data.$translation) {
+      document.querySelector("html").setAttribute("lang", data.$translation.code);
+    }
+
+    // set the document title according to $view.title
+    if (data.$view.title) {
+      document.title = data.$view.title + " | Kirby Panel";
```

Comment:
So far the site title has been used here instead of the hardcoded "Kirby Panel" - I think we should bring that back.